File: Python/Linea.py
from Punto import Punto
class Linea:
    def __init__(self, p1: Punto, p2: Punto):
        self.p1 = p1
        self.p2 = p2

    def __str__(self):
        return f"Línea desde {self.p1.coord_cartesianas} hasta {self.p2.coord_cartesianas()}"

    def dibuja_linea(self):
        print(f"Dibujando una línea desde {self.p1.coord_cartesianas()} hasta {self.p2.coord_cartesianas()}")

punto1 = Punto(1.0, 2.0)
punto2 = Punto(4.0, 6.0)
linea = Linea(punto1, punto2)
print(linea)
linea.dibuja_linea()

File: Java/src/Practica1,2/Linea.java
package Practica1;

public class Linea {
    private Punto p1;
    private Punto p2;

    // Constructor
    public Linea(Punto p1, Punto p2) {
        this.p1 = p1;
        this.p2 = p2;
    }

    @Override
    public String toString() {
        return "Línea desde " + p1.coord_cartesianas() + " hasta " + p2.coord_cartesianas();
    }

 
    public void dibujaLinea() {
        System.out.println("Dibujando una línea desde " + p1.coord_cartesianas()+ " hasta " + p2.coord_cartesianas());
    }

    public static void main(String[] args) {
        Punto punto1 = new Punto(1.0f, 2.0f);
        Punto punto2 = new Punto(4.0f, 6.0f);
        Linea linea = new Linea(punto1, punto2);
        System.out.println(linea);
        linea.dibujaLinea();
    }
}
